packages/@aws-cdk/aws-certificatemanager/lib/dns_validated_certificate_handler/index.js

```diff
@@ -0,0 +1,196 @@
+  }).promise()
+
+  console.log(`Certificate ARN: ${reqCertResponse.CertificateArn}`)
+
+  console.log('Waiting for ACM to provide DNS records for validation...')
+
+  var describeCertResponse
+  let attempt = 0
+  do {
+    // Exponential backoff based on 100ms base
+    await sleep(Math.pow(attempt, 2) * 100)
+    describeCertResponse = await acm.describeCertificate({
+      CertificateArn: reqCertResponse.CertificateArn
+    }).promise()
+  } while (describeCertResponse.Certificate.DomainValidationOptions[0].ValidationStatus !== 'PENDING_VALIDATION')
+
+  const record = describeCertResponse.Certificate.DomainValidationOptions[0].ResourceRecord
+
+  console.log(`Upserting DNS record into zone ${hostedZoneId}: ${record.Name} ${record.Type} ${record.Value}`)
+
+  const changeBatch = await route53.changeResourceRecordSets({
+    ChangeBatch: {
+      Changes: [{
+        Action: 'UPSERT',
+        ResourceRecordSet: {
+          Name: record.Name,
+          Type: record.Type,
+          TTL: 60,
+          ResourceRecords: [{
+            Value: record.Value
+          }]
+        }
+      }]
+    },
+    HostedZoneId: hostedZoneId
+  }).promise()
+
+  console.log('Waiting for DNS records to commit...')
+  await route53.waitFor('resourceRecordSetsChanged', {
+    // Wait up to 5 minutes
+    $waiter: {
+      delay: 30,
+      maxAttempts: 10
+    },
+    Id: changeBatch.ChangeInfo.Id
+  }).promise()
+
+  console.log('Waiting for validation...')
+  await acm.waitFor('certificateValidated', {
+    // Wait up to 5 minutes
+    $waiter: {
+      delay: 30,
+      maxAttempts: 10
+    },
+    CertificateArn: reqCertResponse.CertificateArn
+  }).promise()
+
+  return reqCertResponse.CertificateArn
+}
+
+/**
+ * Deletes a certificate from AWS Certificate Manager (ACM).
+ *
+ * @param {string} arn The certificate ARN
+ */
+const deleteCertificate = async function (arn) {
+  const acm = new aws.ACM()
+
+  console.log(`Deleting certificate ${arn}`)
+
+  await acm.deleteCertificate({
+    CertificateArn: arn
+  }).promise()
+}
+
+/**
+ * Main handler, invoked by Lambda
+ */
+exports.certificateRequestHandler = async function (event, context, callback) {
+  var responseData = {}
+  var physicalResourceId
+  var certificateArn
+
+  try {
+    switch (event.RequestType) {
+      case 'Create':
+        certificateArn = await requestCertificate(event.RequestId, event.ResourceProperties.DomainName, event.ResourceProperties.HostedZoneId)
+        responseData.Arn = physicalResourceId = certificateArn
+        break
+      case 'Delete':
+        physicalResourceId = event.PhysicalResourceId
+        // If the resource didn't create correctly, the physical resource ID won't be the
+        // certificate ARN, so don't try to delete it in that case.
+        if (physicalResourceId.startsWith('arn:')) {
+          await deleteCertificate(physicalResourceId)
+        }
+        break
+      case 'Update':
+        await deleteCertificate(event.PhysicalResourceId)
```

Comment:
I believe this is not necessary. CloudFormation will send a Delete event if after an Update the PhysicalId will have changed.

Reply:
Couldn't hurt, though...

Reply:
Will fix